data-access/nexustiles/model/nexusmodel.py

```diff
@@ -100,6 +136,8 @@ def nexus_point_generator(self, include_nan=False):
     def get_indices(self, include_nan=False):
         if include_nan:
             return list(np.ndindex(self.data.shape))
+        if self.is_multi:
+            return np.argwhere(self.data[0])
```

Comment:
I'm beginning to doubt this after a comment by Vardis during the SDAP meeting today -- if it's possible each variable is masked differently we would need to get the mask separately per-variable. This would have major performance implications for swath datasets so I need to give this more thought.

Reply:
I may not totally understand your concern but, I don't think that's exactly what he meant. For two variables in the same collection, if they came from different instruments and/or have different quality attributes to them it is possible that there is a measurement at index i,j for variable 1 but not at index i,j for variable 2.

The "mask" is already part of the data (in the form of fill values). When read by numpy they get turned into masked arrays where the mask is part of the data array itself. So you don't need to look up a mask based on the variable being read, it's just part of the data

Reply:
The data is formatted such that data[0] == variable 1, data[1] == variable 2, etc. I am only getting the indices for data[0], the first variable because I assumed the mask would be the same across all variables. If that isn't the case I would need to retrieve the mask for the entire data array, which I was trying to avoid because it's *huge* for swath multi-var data.

Reply:
I see, so you're trying to return the indices where valid data exist? That would be dependent on the data variable.

Reply:
Thanks @frankinspace, I will make that change.